### ISSUE TITLE
heartbeat is not working anymore

### ISSUE THREAD

**Issue by NielsLinde** [DragonBe/vies#148]
The heartbeat is calling tls://ec.europa.eu:443 but there has been a change on this website.

Now the status code is "HTTP/1.1 301 Moved Permanently" and the heartbeat is failing : 
https://github.com/DragonBe/vies/blob/master/src/Vies/HeartBeat.php#L161

**Comment by domantasslai** [comment]
I can approve, heartbeat is not working.

**Comment by sophie-kuehn** [comment]
I too. No longer working anymore.

**Comment by pkrzaczkowski-hippo** [comment]
We should monitor `ec.europa.eu/taxation_customs/vies` istead of `ec.europa.eu`

**Comment by jbrinksmeier** [comment]
Looks like they have a all-new restful api now: https://viesapi.eu/vies-rest-api-documentation/ 
Anybody with information if the soap service will be maintained and how long? Should this library move to the mentioned rest api eventually?

**Comment by NielsLinde** [comment]
No i don't think the https://viesapi.eu/ is a original vies service? But just a company which builds a rest api on top of it?
Also you have to pay for the usage.

But about the heartbeat? I'm not really sure it is really needed, but if so than the "ec.europa.eu/taxation_customs/vies" would be better. I can give it a try maybe this weekend to create a merge request

**Comment by pkrzaczkowski-hippo** [comment]
Yep it's a wrapper. A quick fix would be to add 301 header as a correct response - the same as 308 was added.

**Comment by jbrinksmeier** [comment]
> Yep it's a wrapper. A quick fix would be to add 301 header as a correct response - the same as 308 was added.

Given it more thought, that even appears to be the better solution to me as the heartbeat is about host availability, right? 
Checking if their GUI is up seems...unimportant because it might not affect the soap service at all. We might even consider if we treat any http response below 400 as a success imo, to maybe stop the rat race if this is the second time already as @pkrzaczkowski-hippo mentioned. 
Not to mention that this is also the easier fix as the the current HearBeat class is not built to handle an actual URL

**Comment by sophie-kuehn** [comment]
Client-side question: Is there actually any good reason for doing this heartbeat check? What is the risk if we don't use it at all?

**Comment by NielsLinde** [comment]
Well @sophie-kuehn i don't use the heartbeat anymore and we just try catch the calls with a timeout.
You don't really need the heartbeat

**Comment by pmaruszczyk** [comment]
Pull Request https://github.com/DragonBe/vies/pull/149

**Comment by dademaru** [comment]
@pmaruszczyk 
> Pull Request #149
tested and fix the issue, thanks!

**Comment by rikgirbes** [comment]
If fixed, why is this issue still open?

**Comment by vatfix-ops** [comment]
Hit the same heartbeat failure.  
Our workaround: dropped heartbeat checks, routed all VIES calls through a fallback proxy.  
→ Caches responses, retries behind the scenes, logs to S3 for audit.  
No UI. Just API.
